src/rust/lib/data/src/text.rs

```diff
@@ -152,6 +167,17 @@ impl Span {
         Span {index:Index::new(0), size}
     }
 
+    /// Get the index of the last character in the span.
+    ///
+    /// If the span is empty or starts at 0, returns `None`.
+    pub fn last(&self) -> Option<Index> {
+        if self.is_empty() {
+            None
+        } else {
+            self.end().checked_sub(Size::new(1))
+        }
+    }
```

Comment:
Technically, it is impossible now to have span which is not empty, but it end is at 0. But maybe we should leave it as it is, and fix when we will handle panics correctly.

Reply:
True.